### PR TITLE
Remove TFLM from Arduino library download cache

### PR DESCRIPTION
**Pull request by wrighkv1**
Removed 'https://github.com/bcmi-labs/tensorflow_lite_mirror' from the list
Removing TFLM from Arduino library download cache.

### CODE REVIEW

**Comment by per1234** [comment]
Hi @wrighkv1. Although everyone is welcome to submit any library, changes to existing registry entries may only be done with the authorization of a maintainer of the library. Unfortunately, I was unable to verify that this request was authorized.

Please provide proof that this request is authorized. Thanks!

**Comment by wrighkv1** [comment]
Hi [per1234](https://github.com/per1234),

Thanks for the quick response. I am a member of the TensorFlow Lite Micro team. We would like to remove this TFLM library from the Arduino library download cache.

**Comment by advaitjain** [comment]
> Hi @wrighkv1. Although everyone is welcome to submit any library, changes to existing registry entries may only be done with the authorization of a maintainer of the library. Unfortunately, I was unable to verify that this request was authorized.
> 
> Please provide proof that this request is authorized. Thanks!

@per1234, how do we go about finding who the maintainer of the library is? The git history / blame for repositories.txt isn't letting me get to that information and the registry entry that we want to remove (https://github.com/bcmi-labs/tensorflow_lite_mirror) is not a valid repository.

The TensorFlow Lite Micro team (of which both me and @wrighkv1 are members) is maintaining the library at https://github.com/tensorflow/tflite-micro-arduino-examples.

In the current situation, the version of TfLite Micro that is available from the Library Manager is unclear to us.

**Comment by per1234** [comment]
>how do we go about finding who the maintainer of the library is?

Arduino maintains this library (not the source code itself, but the packaging of TensorFlow Lite into a valid Arduino library). "bcmi-labs" is one of the GitHub organizations owned by Arduino.

>the registry entry that we want to remove (https://github.com/bcmi-labs/tensorflow_lite_mirror) is not a valid repository.

It is a valid repository. It is only that it is a private repository. The only reason for it being private is to avoid wasting the time of contributors who might submit issues or pull requests for changes to the mirrored code, not understanding that those contributions must be made to [the upstream repository](https://github.com/tensorflow/tflite-micro).

The contents of [the repository](https://github.com/bcmi-labs/tensorflow_lite_mirror) are of course are no secret, as they are publicly available from the Arduino Library Manager. But the Git data is removed from all the library packages in Library Manager for efficiency, so we do miss the revision control history.

>the version of TfLite Micro that is available from the Library Manager is unclear to us.

I apologize for any confusion. I will discuss this with the relevant people at Arduino.

**Comment by per1234** [comment]
Hi @wrighkv1 and @advaitjain. Before proceeding with the requested removal, I would like to make sure you are fully aware of the implications:

### Loss of Library Manager installation option

The library will be removed completely from the Arduino Library Manager index, which will break any tutorials, etc. which rely on this standard installation method. For example:

- https://github.com/tensorflow/tflite-micro-arduino-examples#arduino-ide
- https://docs.arduino.cc/tutorials/nano-33-ble-sense/get-started-with-machine-learning#ide-setup
- https://learn.sparkfun.com/tutorials/programming-the-sparkfun-edge-with-arduino/all#set-up-for-success
- https://www.digikey.com/en/maker/projects/intro-to-tinyml-part-2-deploying-a-tensorflow-lite-model-to-arduino/59bf2d67256f4b40900a3fa670c14330#:~:text=Install%20TensorFlow%20Lite%20Arduino%20Library

### Loss of availability of Arduino's releases

It might be challenging for those interested in using an existing project which pinned one of the releases of the library formerly distributed by Arduino to source that same version elsewhere, since `tensorflow/tflite-micro-arduino-examples` does not have any tags and it seems not so simple for the average user to derive the library directly from the source in the `tensorflow/tflite-micro` repository.

The alternative will be for them to use a version of the library the project was not validated for.

### Arduino provides precompiled releases of the library

Arduino's packaging infrastructure for the library produces [precompiled](https://arduino.github.io/arduino-cli/latest/library-specification/#precompiled-binaries) releases which dramatically reduce what would otherwise be a lengthy compilation time for the user's sketches that use the library.

**Comment by ladyada** [comment]
for many adafruit libraries we had some questions, so this is good timing - you can make github repos that are public but do not have Issues or forks enabled (without forking you wont get PRs submitted), great for keeping things quiet :)

image

**Comment by per1234** [comment]
Thanks @ladyada! That "**Allow forking**" setting is very interesting to me for use in a couple other public mirror repositories:

- https://github.com/arduino-libraries/Arduino_BHY2
- https://github.com/arduino-libraries/Arduino_BHY2Host

Unfortunately, I looked into their settings and didn't see this "**Allow forking**" setting:

image

I do see it in a private repository though. Was the screenshot you shared from a private repo?

GitHub's docs seem to indicate the feature is for private repos:

https://docs.github.com/en/repositories/managing-your-repositorys-settings-and-features/managing-repository-settings/managing-the-forking-policy-for-your-repository

For preventing PRs, we are currently using [the "Interaction Limits" setting](https://docs.github.com/en/communities/moderating-comments-and-conversations/limiting-interactions-in-your-organization). The only problem with this approach is it has a maximum period of 6 months, so we must remember to periodically cycle the setting to avoid it expiring.

**Comment by advaitjain** [comment]
Thanks a lot for the detailed replies!

After giving it some more thought we came to the conclusion that we would still like to continue with removing the library manager installation for Tensorflow Lite Micro.

**Comment by per1234** [comment]
@wrighkv1 @advaitjain I am confirming your request.

Unlike submissions, this type of request is still handled manually. The Library Manager index maintainer occasionally comes through to clear all the outstanding requests. They will merge this pull request when it is done. You will get a notification from GitHub at that time.

**Review by umbynos — APPROVED** [review]
Done:
```
2022/08/31 16:21:41 Loaded 5121 libraries from DB
Removing Arduino_TensorFlowLite
warning: Library Git clone folder /home/ubuntu/workspace/gitclones/github.com/bcmi-labs/tensorflow_lite_mirror not present
Success!
```

**Pull request merged**
